### PR TITLE
WIP: Add reconciliation of kibana cluster uuid

### PR DESCRIPTION
**Pull request by periklis**
### Description
This PR add reconciliation of a pre-generated UUID for kibana clusters. This UUID is stored as a ConfigMap and in turn allows Kibana to re-use it across restarts.

/cc @blockloop 
/assign @jcantrill 

### Links
- Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1933978

### CODE REVIEW

File: internal/k8shandler/kibana/reconciler.go [hunk 1]
```diff
@@ -89,8 +90,22 @@ func Reconcile(requestCluster *kibana.Kibana, requestClient client.Client, esCli
 		return err
 	}
 
+	clusterUUID := uuid.New().String()
```
Comment (jcantrill) [inline internal/k8shandler/kibana/reconciler.go:93]: not sure how strict uuid is to Kibana, but a more predictable pattern so we do not have to stash it someplace:
* md5 the name or ns/name
* if there is a way to predictably generate a uuid from a seed value

File: internal/k8shandler/kibana/reconciler.go [hunk 2]
```diff
@@ -499,7 +515,7 @@ func newKibanaPodSpec(cluster *KibanaRequest, elasticsearchName string, proxyCon
 	kibanaContainer := NewContainer(
 		"kibana",
 		kibanaImage,
-		v1.PullIfNotPresent,
+		v1.PullAlways,
```
Comment (jcantrill) [inline internal/k8shandler/kibana/reconciler.go:518]: will need to revert

File: internal/k8shandler/kibana/reconciler.go [hunk 3]
```diff
@@ -519,10 +535,19 @@ func newKibanaPodSpec(cluster *KibanaRequest, elasticsearchName string, proxyCon
...
+		},
+		{
+			Name:  "PATH_DATA",
+			Value: "/data",
+		},
 	}
 
 	kibanaContainer.VolumeMounts = []v1.VolumeMount{
 		{Name: "kibana", ReadOnly: true, MountPath: "/etc/kibana/keys"},
+		{Name: "kibana-state", ReadOnly: false, MountPath: "/data"},
```
Comment (jcantrill) [inline internal/k8shandler/kibana/reconciler.go:550]: we should mount in a single file to the directory

**Review by jcantrill — COMMENTED** [review]
/hold
/approve

**Comment by periklis** [comment]
I close this PR as it is not providing a solution towards my investigation for https://bugzilla.redhat.com/show_bug.cgi?id=1933978.

TL;DR; The proposed fix to make kibana's uuid stable and seeded from a known key per suggestion in opendistro-for-elasticsearch/security-kibana-plugin#460 is not applying to OD's kibana tenant index migrations. The issue is more of the sort of pending upgrade of OD plugin to v0.10.1.2 and ES 6.8.6.

(cc @jcantrill @ewolinetz)

/close
